addons/sale_timesheet/models/sale_order.py

```diff
@@ -51,10 +51,11 @@ def _compute_project_ids(self):
     def _action_confirm(self):
         """ On SO confirmation, some lines should generate a task or a project. """
         result = super(SaleOrder, self)._action_confirm()
-        self.mapped('order_line').sudo().with_context(
-            default_company_id=self.company_id.id,
-            force_company=self.company_id.id,
-        )._timesheet_service_generation()
+        for line in self.mapped('order_line'):
+            line.sudo().with_context(
+                default_company_id=line.company_id.id,
+                force_company=line.company_id.id,
+            )._timesheet_service_generation()
```

Comment:
Keep the lines grouped when possible ;).  The company of the lines is the same than the order one as it is a related field.
```suggestion
        for order in self:
            order.mapped('order_line').with_context(
                default_company_id=order.company_id.id,
                force_company=order.company_id.id,
            )._timesheet_service_generation()
```